### PR TITLE
chore: add grafonnet dashboards support

### LINKED ISSUE #3763: Maintainable grafana dashboard
try to refactor dashboard with https://github.com/istio/istio/blob/395b9bb706a4fcb8633beac188b64707d7e7a1f1/manifests/addons/dashboards/README.md to make it maintainable

_Originally posted by @zirain in https://github.com/envoyproxy/gateway/issues/3746#issuecomment-2208137341_

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

**What this PR does / why we need it**:

- merge `envoy-gateway-resources` and `envoy-proxy-resources` dashboards using Grafonnet
- highlight doc to suggest using Grafonnet to build out new dashboards
- for all the old dashboard, we keep it as legacy dashboards

**Which issue(s) this PR fixes**:

Fixes #3763

### CODE REVIEW

File: charts/gateway-addons-helm/dashboards/lib/g.libsonnet
Comment (zirain) [inline charts/gateway-addons-helm/dashboards/lib/g.libsonnet:1]: prefer to move these out of charts folder, it's more like source code.

Comment (shawnh2) [reply charts/gateway-addons-helm/dashboards/lib/g.libsonnet:1]: dashboard use relative path to import things under `lib` folder, like `local g = import 'lib/g.libsonnet';`. move lib elsewhere could make the import path a little more messy, so should be better leave it here.

I also include this `lib` folder in `.helmignore`, so the helm will ignore it, won't be able to package it into chart.

**Review by Xunzhuo — APPROVED** [review]

**Comment by Xunzhuo** [comment]
Let's get this in

**Comment by Xunzhuo** [comment]
@shawnh2 plz fix the lint errors

**Comment by shawnh2** [comment]
> @shawnh2 plz fix the lint errors

done, PTAL

**Review by Xunzhuo — APPROVED** [review]

**Review by zirain — APPROVED** [review]

**Pull request merged**